broadinstitute/single_cell_portal_core
# Show author-computed differential expression data in table (SCP-5190)

This lets users see DE computed by authors in the table, completing the UI more fully described in #1806.

Previously, users could only select which groups to compare.  Now, they can see author-computed results in the DE table.

Here's how it looks:

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/8e12c28a-f3d5-4b30-a19a-3e52cafdc231

The [richer SCP demo](https://drive.google.com/file/d/18lL2tMJSdyxQ5KEg8mpJToG6qpWitO2J/view?t=30s) and discussion are also available.

To manually test:
1.  Pull branch
2. Go to "Human milk - differential expression" study. If on staging after this is merged, that's [SCP303](https://singlecell-staging.broadinstitute.org/single_cell/study/SCP303/human-milk-differential-expression).
3. Copy processed author DE data to your bucket, e.g. `BUCKET_ID="<your_bucket_ID, e.g. fc-febd4c65-881d-497f-b101-01a7ec427e6a>"; gsutil -m cp test/test_data/human_author_de/* gs://$BUCKET_ID/_scp_internal/differential_expression/`
4. Select annotation "General_Celltype"
5.  Click "Differential expression"
6.  Select "B cells" in 1st menu, and "rest" in 2nd menu
7. Confirm data appears in DE table
8. Select "dendritic cells" in 2nd menu
9. Confirm data appears in DE table 

This satisfies SCP-5190.
